x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -42,17 +60,175 @@
  * shard files and what it stored in the snapshot the {@link BlobStoreIndexShardSnapshot} is used to map a physical file name as expected by
  * Lucene with the one (or the ones) corresponding blob(s) in the snapshot.
  */
-public class SearchableSnapshotDirectory extends BaseSearchableSnapshotDirectory {
+public class SearchableSnapshotDirectory extends BaseDirectory {
 
-    SearchableSnapshotDirectory(final BlobStoreIndexShardSnapshot snapshot, final BlobContainer blobContainer) {
-        super(blobContainer, snapshot);
+    private final BlobStoreIndexShardSnapshot snapshot;
+    private final BlobContainer blobContainer;
+    private final SnapshotId snapshotId;
+    private final IndexId indexId;
+    private final ShardId shardId;
+    private final LongSupplier statsCurrentTimeNanosSupplier;
+    private final Map<String, IndexInputStats> stats;
+    private final CacheService cacheService;
+    private final boolean useCache;
+    private final Path cacheDir;
+    private final AtomicBoolean closed;
+
+    public SearchableSnapshotDirectory(
+        BlobContainer blobContainer,
+        BlobStoreIndexShardSnapshot snapshot,
+        SnapshotId snapshotId,
+        IndexId indexId,
+        ShardId shardId,
+        Settings indexSettings,
+        LongSupplier currentTimeNanosSupplier,
+        CacheService cacheService,
+        Path cacheDir
+    ) {
+        super(new SingleInstanceLockFactory());
+        this.snapshot = Objects.requireNonNull(snapshot);
+        this.blobContainer = Objects.requireNonNull(blobContainer);
+        this.snapshotId = Objects.requireNonNull(snapshotId);
+        this.indexId = Objects.requireNonNull(indexId);
+        this.shardId = Objects.requireNonNull(shardId);
+        this.stats = ConcurrentCollections.newConcurrentMapWithAggressiveConcurrency();
+        this.statsCurrentTimeNanosSupplier = Objects.requireNonNull(currentTimeNanosSupplier);
+        this.cacheService = Objects.requireNonNull(cacheService);
+        this.cacheDir = Objects.requireNonNull(cacheDir);
+        this.closed = new AtomicBoolean(false);
+        this.useCache = SNAPSHOT_CACHE_ENABLED_SETTING.get(indexSettings);
+    }
+
+    public BlobContainer blobContainer() {
+        return blobContainer;
+    }
+
+    public SnapshotId getSnapshotId() {
+        return snapshotId;
+    }
+
+    public IndexId getIndexId() {
+        return indexId;
+    }
+
+    public ShardId getShardId() {
+        return shardId;
+    }
+
+    public Map<String, IndexInputStats> getStats() {
+        return Collections.unmodifiableMap(stats);
+    }
+
+    @Nullable
+    public IndexInputStats getStats(String fileName) {
```

Comment:
Comes from the now removed CacheDirectory and made `public` as cache classes are in a different package